lib/mongoid/extensions/big_decimal.rb

```diff
@@ -67,21 +63,24 @@ def demongoize(object)
         # @param [ Object ] object The object to Mongoize
         #
         # @return [ String | BSON::Decimal128 | nil ] A String or Decimal128
-        #   representing the object or nil.
+        #   representing the object or nil. String if Mongoid.map_big_decimal_to_decimal128
+        #   is false, BSON::Decimal128 otherwise.
         def mongoize(object)
-          unless object.nil?
+          return if object.nil?
+          return if object.is_a?(String) && object.blank?
```

Comment:
probably fine to just do `return if object.blank?` here. Skipping blank Array/Hash is not a problem IMHO.

Reply:
sure